packages/@aws-cdk/aws-pipes-sources-alpha/lib/deadLetterConfig.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * A DeadLetterConfig object that contains information about a dead-letter queue configuration.
+ */
+export interface DeadLetterConfigParameters {
+  /**
+   * The ARN of the specified target for the dead-letter queue,
+   *
+   * For Amazon Kinesis stream and Amazon DynamoDB stream sources, specify either an Amazon SNS topic or Amazon SQS queue ARN.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-deadletterconfig.html#cfn-pipes-pipe-deadletterconfig-arn
+   * @default no dead letter queue
+   */
+  readonly arn?: string;
```

Comment:
It might be more explicit to have construct interfaces instead of the `arn` there, especially given it accepts both SQS and SNS? From a user perspective, they can just either use the queue construct they just created, or retrieve it with `.fromQueue/TopicArn()`:

```suggestion
  readonly target?: IQueue | ITopic;
```

Reply:
@nmussy I made this change, but now there is a dependency between the DLQ and the pipe (see below). I'm not quite sure where to add the dependency. Maybe in `aws-pipes-alpha/pipes.ts` ? Let me know if you have any ideas. 

```
Deployment failed: Error: The stack named aws-cdk-pipes-sources-dynamodb-stream failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Resource handler returned message: "Resource of type 'AWS::Pipes::Pipe' with identifier 'Pipe7793F8A1-B5m3oRxh5ZKt' did not stabilize. Status Reason is Input parameter is invalid from the request due to : Error occurred while sending message to SQS queue: arn:aws:sqs:us-east-1:REDACED:aws-cdk-pipes-sources-dynamodb-stream-DlqQueueFDA42DA7-LKdmBtvZnzrn" (RequestToken: 63c085a3-c417-a234-e570-343c179f1333, HandlerErrorCode: NotStabilized)
```

Reply:
hmm, something is wrong with my code. The permissions aren't being added to the role. Calling it for the night but I'll try to figure this out.

Reply:
If it's an issue caused by interface constructs or the union type, a good compromise might be to go back to an interface and separate the ARN props:
```ts
export interface DeadLetterConfig {
  /**
   * The ARN of the target Amazon SQS queue.
   * 
   * At least one of {@link queueArn} and {@link topicArn} must be provided.
   * 
   * @default - not an SQS dead letter queue
   */
  readonly queueArn?: string;

  /**
   * The ARN of the target Amazon SNS topic.
   * 
   * At least one of {@link queueArn} and {@link topicArn} must be provided.
   * 
   * @default - not an SNS dead letter queue
   */
  readonly topicArn?: string;
}
```

Reply:
What are your thoughts on granting permissions to the queue/topic outside of this construct? i.e., specify in the README and the integration test the role must have permission to push to them.

Here's what I have now... It's not working but I'll keep investigating.
https://github.com/aws/aws-cdk/pull/29476/files#diff-4bb7f8b01c48466253fab15c1b3492a1e240a2fc1a272b48a90c01534636431eR180-R184

Reply:
> What are your thoughts on granting permissions to the queue/topic outside of this construct? i.e., specify in the README and the integration test the role must have permission to push to them.

That would be less than ideal, I would expect an L2 construct to handle that for me

Reply:
yeah... let me figure out why these permissions aren't being applied.

Reply:
Let me know if you want me to take a look 👍

Reply:
I had forgotten to build `aws-pipes-alpha/lib/source.ts` 🤦🏼‍♂️